src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -823,9 +825,26 @@ private void mapVendorDetail(List<CompletableFuture<?>> futures, VendorDetail ve
       vendorDetail.setReferenceNumbers(referenceNumbers);
     });
 
-    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_ACCOUNT))
+    //[MODGOBI-132]
+    //match subAccount send to GOBI with the accountNo established in Organisation, if match found then map otherwise map  the received subaccount as accountNo.
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR))
       .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> vendorDetail.setVendorAccount((String) o))
+        .thenAccept(o -> {
+          if (o != null) {
+          Organization organization = (Organization) o;
+          Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_ACCOUNT))
+          .ifPresent(vendorAccountfield -> {
+              if (!HelperUtils.extractSubAccount(organization.getAccounts().get(0).getAccountNo()).equals(vendorAccountfield.resolve(doc).join())) {
```

Comment:
NPE can be possible organization.getAccounts().get(0).getAccountNo()

Reply:
Also don't include int equals complex logic lie vendorAccountfield.resolve(doc).join()
if it fail then will be difficult to understand a root cause